src/mongoc/mongoc-topology.c

```diff
@@ -351,29 +359,23 @@ mongoc_topology_select (mongoc_topology_t         *topology,
          topology->stale = false;
       }
 
-      /* until we find a server or timeout */
-      for (;;) {
```

Comment:
Which looping?
There isn't a automatic retry-loop anymore.

I do believe we should set topology->stale before the RETURN (NULL) case though, so subsequent user-triggered explicit retries has a chance of succeeding by triggering a rescan.
Although - I guess that is replaced with the blacklisting now.